netutils/netinit/netinit.c

```diff
@@ -344,66 +340,239 @@ static void netinit_set_macaddr(void)
 #endif
 
 /****************************************************************************
- * Name: netinit_set_ipaddrs
+ * Name: netinit_set_ipv4addrs
  *
  * Description:
- *   Setup IP addresses.
- *
- *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
- *   to any user provided value is asking for trouble.
+ *   Setup IPv4 addresses.
  *
  ****************************************************************************/
 
 #if defined(NETINIT_HAVE_NETDEV) && !defined(CONFIG_NET_6LOWPAN) && ! \
-    defined(CONFIG_NET_IEEE802154)
-static void netinit_set_ipaddrs(void)
+    defined(CONFIG_NET_IEEE802154) && defined(CONFIG_NET_IPv4)
+static inline void netinit_set_ipv4addrs(void)
 {
-#ifdef CONFIG_NET_IPv4
   struct in_addr addr;
+#ifdef CONFIG_FSUTILS_IPCFG
+  struct ipv4cfg_s ipv4cfg;
+  int ret;
+#endif
 
-  /* Set up our host address */
+#ifdef CONFIG_FSUTILS_IPCFG
+  /* Attempt to obtain IPv4 address configuration from the IP configuration
+   * file.
+   */
 
-#ifndef CONFIG_NETINIT_DHCPC
-  addr.s_addr = HTONL(CONFIG_NETINIT_IPADDR);
+  ret = ipcfg_read(NET_DEVNAME, (FAR struct ipcfg_s *)&ipv4cfg, AF_INET);
+#ifdef CONFIG_NETUTILS_DHCPC
+  if (ret >= 0 && ipv4cfg.proto != IPv4PROTO_NONE)
```

Comment:
No, that is impossible in a modular, layered architecture.  User-space applications must be completely ignorant of the underlying media and hence must never make any assumptions about media properties, whether or not it is eraseable or what the erased value it.  Reading from an erased FLASH would fail for the base the binary file:  The address family of 0xffff would be invalid; it would not match AF_INET or AF_INET6  The hdr.next field would be 0xff so that might go off into never-never-land.  I suppose there should be a sanity check on the hdr.next field too.

I think your PR #419 addresses the versioning issues.

Reply:
Resolving this conversation because the question, while important, does not apply to this PR.

Reply:
> he hdr.next field would be 0xff so that might go off

yes - the version, can catch this. May be it should  be checked first.